### ISSUE TITLE
Tagger fails on every search instance

### ISSUE THREAD

**Issue by erikkristiaan** [Marekkon5/beatporttagger#15]
Started occurring after beatport did a small UI change. Screenshots attached.

![image](https://user-images.githubusercontent.com/8892722/107427418-60482180-6aef-11eb-8f96-0780ceb58f7c.png)
![image](https://user-images.githubusercontent.com/8892722/107427497-79e96900-6aef-11eb-9a32-1c4612222092.png)

I imagine it might have something to do with beautifulsoup running into problems finding webpage elements with the UI changes but I haven't looked at the code in depth so I can't say for sure.

Love the app by the way.

**Comment by Marekkon5** [comment]
Yes, it happened 2 days ago also, but it was reverted overnight. We are waiting whether the change will be permanent or they will revert again.

**Comment by erikkristiaan** [comment]
Great, thanks for the update!

**Comment by atarbig** [comment]
My friend wrote:
It does look like it's a problem on the Beatport site:
This works:
https://www.beatport.com/genre/breaks/9/tracks?per-page=150
However, if you try to change the sort order it breaks:
https://www.beatport.com/genre/breaks/9/tracks?per-page=150&=&sort=release-desc
If makes a request and specifies the sort order so that's probably why it's getting nothing back.. It looks like release-desc is the default maybe try off it. In any case, this is a problem on the beatport website

Maybe it will be useful for you

**Comment by Marekkon5** [comment]
The tagger doesn't use the sorting. It only has query when searching.

**Comment by frankiebones2016** [comment]
Hello I am having the same problem. I am using on windows 8.1. Every search has failed.

**Comment by atarbig** [comment]
hmm, on beatport search works
https://www.beatport.com/search/tracks?q=crazibiza+roller+disco
https://www.beatport.com/search?q=crazibiza+roller+disco
Waiting for it to work in the tagger

**Comment by Marekkon5** [comment]
@atarbig 
Did you try running from latest source?

**Comment by atarbig** [comment]
i donwnloaded last source, see debug.log:
[0212/121059.676:ERROR:icu_util.cc(133)] Invalid file descriptor to ICU data received.
beatporttagger.exe no start (from folder dist)
python 3.7.9 and 3.8.6 tested (windows10)

**Comment by frankiebones2016** [comment]
[0212/022150.169:ERROR:mf_helpers.cc(14)] Error in dxva_video_decode_accelerator_win.cc on line 534
Tested on windows 8.1 with Beatport.Tagger.exe. Fails to identify search.

**Comment by Marekkon5** [comment]
On latest source please run using python, don't try to package it - it is really broken.

`pip install -r requirements.txt`
`python beatporttagger.py`

**Comment by atarbig** [comment]
python beatporttagger.py works but still nothing found;
i test with last code https://github.com/Marekkon5/beatporttagger/archive/1.1.7.zip

**Comment by Marekkon5** [comment]
@atarbig 

Beatport is still doing some changes, the search is really unstable and broken even on the website.

Also 1.1.7 isn't the latest, download main source code: https://github.com/Marekkon5/beatporttagger/archive/main.zip

**Comment by atarbig** [comment]
this code work, and thank you for info, will wait news

**Comment by bascurtiz** [comment]
Just to confirm, I reached out to Beatport: they could tell me they are implementing the new API.
Seems sort function is broken, which I made them aware of: https://youtu.be/Ppp1c3B3E7Q

Note that in above video (that was yesterday) it did find the track example:
https://www.beatport.com/search?q=alex+morelli+crimson+idol
When I search that right now, it's not showing up.

Also, the other example can only be found as release, but not showing up under track results:
https://www.beatport.com/search?q=crazibiza+roller+disco

Since Beatport Tagger is searching for track matches, not release matches, it cant find it, so that makes sense.

TL;DR:
Beatport seems to be busy updating their API and broke sort function on their site.
Also search results are inconsistent at the moment.
Not really something we can do about, until they fixed their sh*t.

**Comment by atarbig** [comment]
Last year, the beatport was already broken and there was a problem with the search for almost a month. They even closed the site for a few days. After that they promised to make a backup of the site and in case of an accident it can be restored But it seems they saved money.
I also noticed that the search for new tracks in the store does not work immediately. For example, on traxsource in about 8 hours. On the Beatport too. I think this is how they update the track database. The fastest update on junodownload - they have servers in the UK and releases appear earlier, and the search for new tracks is about 6 hours earlier than US stores. There will be time to make a junodownload tagger, then it will be a complete kit. If one store is down, you can use another one. Right now we have traxsource only. Will be wait Beatport.

**Comment by Marekkon5** [comment]
I've posted a new release, which should improve the matching, however it's still bad because of the Beatport issues, not the tagger.
